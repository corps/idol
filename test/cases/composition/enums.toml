[EnumOne]
enum = ["one"]

[EnumTwo]
enum = ["two"]

[EnumOneTwo]
is_a = ["EnumOne", "EnumTwo"]
variance = "Contravariant"

[EnumOneWithCo]
is_a = ["EnumOneTwo", "EnumOne"]