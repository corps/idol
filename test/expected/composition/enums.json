{
  "EnumOne": {
    "enum": [
      "one"
    ],
    "fields": {},
    "is_a": [],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "EnumOneTwo": {
    "enum": [
      "one",
      "two"
    ],
    "fields": {},
    "is_a": [],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "EnumOneWithCo": {
    "enum": [],
    "fields": {},
    "is_a": [
      "enums.EnumOne"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "EnumTwo": {
    "enum": [
      "two"
    ],
    "fields": {},
    "is_a": [],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  }
}
